src/mlpack/core/data/dictionary_encoding.hpp

```diff
@@ -0,0 +1,154 @@
+  void CreateMap(std::string& input, TokenizerType tokenizer);
+  /*
+  * Default Constructor
+  */
+  DictionaryEncoding() {}
+  /*
+  * Copy Constructor.
+  */
+  DictionaryEncoding(const DictionaryEncoding& oldObject);
+  /*
+  * Move Constructor.
+  */
+  DictionaryEncoding(DictionaryEncoding&& oldObject) = default;
+  /*
+  * Move Assignment Operator.
+  */  
+  DictionaryEncoding& operator= (DictionaryEncoding&& oldObject) = default;
+  /*
+  * Assignment Operator.
+  */
+  DictionaryEncoding operator= (const DictionaryEncoding& oldObject);
+  /**
+  * A function to reset the mapping that is clear all the encodings
+  */
+  void Reset();
+
+  /**
+  * A fucntion to encode given array of strings using a particular delimiter,
+  * providing custom rule for tokenization.
+  *
+  * For example 
+  * Vector is :
+  * [hello@wow, wow@hello@good] would be encoded using '@' as delimiter as 
+  * [1 2 0, 2 1 3] 
+  * The function paddes 0 to maintain same sizes across all the rows
+  * User may also provide their custom tokenization rule.
+  *
+  * @param input Vector of strings.
+  * @param output Output Matrix to store encoded results (sp_mat or mat).
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename MatType, typename TokenizerType>
+  void Encode(const std::vector<std::string>& input,
+              MatType& output, TokenizerType tokenizer);
+
+  /**
+  * A function to encode given array of strings using a particular delimiter,
+  * with custom tokenization.
+  *
+  * For example 
+  * Vector is :
+  * [hello@wow, wow@hello@good] would be encoded using '@' as delimiter as 
+  * [1 2 , 2 1 3] 
+  * The function does not paddes 0 in this case.
+  *
+  * @param input Vector of strings.
+  * @param output Vector of vectors to store encoded results.
+  * @param tokenizer A function that accepts a boost::string_view as
+  *                  an argument and returns a token.
+  * This can either be a function pointer or function object or a lamda
+  * function. Its return value should be a boost::string_view, a token.
+  *
+  * Definiation of function should be of type
+  * boost::string_view fn()(boost::string_view& str)
+  *
+  */
+  template<typename TokenizerType>
+  void Encode(const std::vector<std::string>& input,
+              std::vector<std::vector<size_t>>& output,
+              TokenizerType tokenizer);
+
+  //! Modify the originalStrings.
+  std::deque<std::string>& OriginalStrings() { return originalStrings; }
+
+  //! Return the originalStrings.
+  const std::deque<std::string>& OriginalStrings() const
+      { return originalStrings; }
+
+  //! Return the Mappings
```

Comment:
I guess it's the mappings (the first character is lowercase).

Reply:
Yes :)